contents/blog/clickhouse-vs-redshift.md

```diff
@@ -0,0 +1,150 @@
+
+Because Redshift is a multipurpose analytical database, its customers are wide-ranging, including Nasdaq, PBS, McDonald's, Roche, and Merck. 
+
+## Architecture
+
+### Redshift’s (ultra-complex) architecture
+
+Redshift has a particularly complex architecture, which is a byproduct of Amazon’s enormous preexisting infrastructure, and Redshift’s general goal of being an all-purpose database. 
+
+Redshift’s infrastructure tagline is that it's a massively parallel, shared-nothing columnar architecture. “Shared-nothing” means that individual Redshift clusters don’t have overlapping data, and “massively parallel” indicates that Redshift splits computation between nodes. 
+
+On paper, these are standard premises behind an OLAP database. Amazon Redshift uses a leader node (sometimes called a master node in other systems) that acts as a load balancer, handling transactions between SQL clients and compute nodes. 
+
+Compute nodes, through some complexity discussed below, transact with storage that is often built on AWS S3, though support for other storage solutions, like RDS, has been added recently. 
+
+![Redshift Simplfiied.png](../images/blog/clickhouse-vs-redshift/Redshift_Simplfiied.png)
+
+Redshift becomes a bit more complex once we dive into its multiple layers of computation optimizations, which are both organizational and bare-metal hardware. 
+
+The first is Redshift Spectrum, which refers to an army of EC2 machines that compute nodes can recruit to split a single query into parallel computations. More accurately, Redshift Spectrum is a layer between compute nodes and storage – while compute nodes divvy up queries via the leader node, Spectrum splits a single query’s computation between many machines. 
+
+![Amazon Redshift Spectrum.png](../images/blog/clickhouse-vs-redshift/Amazon_Redshift_Spectrum.png)
+
+The easiest way to describe this setup is to walk through a single query. Imagine a business intelligence (BI) tool that wants to learn the total sum of sales across a company’s history. In order: 
+
+1. A leader node accepts the query and delegates it to a compute node.
+2. A compute node employs Amazon Redshift Spectrum to execute the summation.
+3. Thousands of machines in Amazon Redshift Spectrum begin adding up multiple months of sales data each, returning those aggregations to the compute node.
+4. The compute node adds all the micro-aggregations together, calculating a total aggregation across all data. 
+5. The compute node returns that value to the leader node. 
+6. The leader node returns that value to the BI tool. 
+
+Things get extra complicated, and extra efficient, with Amazon’s 2019 release of AQUA (Advanced Query Accelerator). AQUA evolved because SSD bandwidth increased immensely over time, but more storage doesn’t inherently mean faster queries, especially when the bottleneck is now between the memory-compute bus. 
```

Comment:
```suggestion
Things get extra complicated and efficient, with Amazon’s 2019 release of AQUA (Advanced Query Accelerator). AQUA evolved because SSD bandwidth increased immensely over time, but more storage doesn’t inherently mean faster queries, especially when the bottleneck is now between the memory-compute bus. 
```